### PR TITLE
Remove usage of `root_path` and further decouple everything

### PR DESCRIPTION
**Pull request by zachguo**
This PR should cover most of the small changes needed, before the upcoming bigger change of directory structure for #395 .

The upcoming "bigger change" will move all core folders such as `generated`, `constants`, `module`, `ovl_util`, `dumps` and `sql_commands` into one folder `cobra_core` (or another name you maintainers pick, maybe `naja`?) The bigger change would cause a lot of changed files in git, making it hard to review, so I split some small changes into this PR to make it easier to review.

In addition to the existing test suites, I manually tested my common workflow of extracting JWE2 FDBs and PNGs using [a zip of my fork](https://github.com/zachguo/cobra-tools/archive/refs/heads/master.zip). However, I haven't used anything related to `manis`, `ms2`, or the Blender plugin, I don't know how to properly test them. I need help from someone to test them out to make sure everything still works properly.

### CODE REVIEW

File: ovl_util/logs.py [hunk 1]
```diff
@@ -11,14 +11,15 @@
 from typing import TextIO
 
 from ovl_util.config import load_config
-from root_path import root_dir
+# TODO: log_dir and config_path used in this file should be parameterized instead of hardcoded based on root_dir
+root_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), ".."))
```
Comment (HENDRIX-ZT2) [inline ovl_util/logs.py:15]: also

File: source/formats/manis/__init__.py [hunk 2]
```diff
@@ -307,7 +306,8 @@ def parse_keys(self, target=None):
             if target and mani_info.name != target:
                 continue
             # acro debug keys
-            dump_path = os.path.join(root_path.root_dir, "dumps", f"{mani_info.name}_keys.txt")
+            pkg_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', '..', '..'))
```
Comment (HENDRIX-ZT2) [inline source/formats/manis/__init__.py:309]: also

**Review by HENDRIX-ZT2 — CHANGES_REQUESTED** [review]
see comments on path formatting

File: generated/formats/manis/__init__.py [hunk 3]
```diff
@@ -308,7 +307,8 @@ def parse_keys(self, target=None):
             if target and mani_info.name != target:
                 continue
             # acro debug keys
-            dump_path = os.path.join(root_path.root_dir, "dumps", f"{mani_info.name}_keys.txt")
+            pkg_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', '..', '..'))
```
[the remark this answers is not in the document]
Comment (zachguo) [reply generated/formats/manis/__init__.py:310]: For accessing `dumps` and `sql_commands`, I planned to use [`importlib.resources`](https://docs.python.org/3/library/importlib.resources.html#module-importlib.resources). However, this requires proper bundling of package data, which I haven't touched yet but will eventually do.

The deprecation of `root_path` was done in a way to minimize the number of lines being changed, for easier review.

File: ovl_util/logs.py (hunk 1 above)
Comment (zachguo) [reply ovl_util/logs.py:15]: Take a look at the `# TODO: ` comment above this line.

File: ovl_util/auto_updater.py [hunk 4]
```diff
@@ -9,9 +9,10 @@
 import logging
 import subprocess
 
-from root_path import root_dir
 from ovl_util.logs import ANSI
 
+root_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), ".."))
```
[the remark this answers is not in the document]
Comment (zachguo) [reply ovl_util/auto_updater.py:14]: I actually plan to move this script to the root directory. (along with some other changes to break `ovl_util` even more)

This script is sort of global, touching `pyproject.toml`, covering both core and GUI, and  technically can be executed as a standalone script.

[the remark this answers is not in the document]
Comment (zachguo) [reply ovl_util/auto_updater.py:14]: I can do those "breaking it up" changes now in this PR. I thought those changes were more akin to the next PR conceptually.

File: modules/formats/FDB.py [hunk 5]
```diff
@@ -41,7 +40,8 @@ def _get_data(self, file_path):
 
 	@staticmethod
 	def open_command(f):
-		command_path = os.path.join(root_dir, "sql_commands", f"{f}.sql")
+		pkg_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', '..'))
```
[the remark this answers is not in the document]
Comment (zachguo) [reply modules/formats/FDB.py:43]: https://github.com/OpenNaja/cobra-tools/pull/398#discussion_r1777800145

File: source/formats/manis/__init__.py (hunk 2 above)
Comment (zachguo) [reply source/formats/manis/__init__.py:309]: https://github.com/OpenNaja/cobra-tools/pull/398#discussion_r1777800145

**Review by HENDRIX-ZT2 — APPROVED** [review]

**Comment by zachguo** [comment]
I'll rebase the PR this weekend.

**Pull request merged**